protocol/messenger_communities.go

```diff
@@ -338,7 +392,14 @@ func (m *Messenger) handleCommunitiesSubscription(c chan *communities.Subscripti
 					return
 				}
 				if sub.Community != nil {
-					publishOrgAndDistributeEncryptionKeys(sub.Community)
+					if sub.Community == nil {
+						continue
+					}
+					// NOTE: because we use a pointer here, there's a race condition where the community would be updated before it's compared to the previous one.
+					// This results in keys not being propagated as the copy would not see any changes
+					communityCopy := sub.Community.CreateDeepCopy()
```

Comment:
Perhaps instead of having a Community pointer in the subscription, we could use a Community ID. This way, we can re-fetch the latest version. This solution, together with the recently introduced `map[communityID, mutex]`, will ensure the latest version of the Community is used and will mitigate race conditions.